### PR TITLE
Update docker go16

### PR DESCRIPTION
**Pull request by ianzhang366**

### CODE REVIEW

File: Dockerfile.prow [hunk 1]
```diff
@@ -2,16 +2,11 @@
 # and a workspace (GOPATH) configured at /go.
 FROM registry.access.redhat.com/ubi8/ubi-minimal:latest
```
Comment (xiangjingli) [inline Dockerfile.prow:3]: should we also use this image as the base

`FROM  registry.ci.openshift.org/open-cluster-management/builder:go1.16-linux-amd64`

Comment (ianzhang366) [reply Dockerfile.prow:3]: the prow will automatically replace it for us, the Dockefile is more suitable for the local tests now...

**Pull request merged**